tools/harness-simulation/harness/Thread_Harness/Sniffer/SimSniffer.py

```diff
@@ -0,0 +1,226 @@
+        for i in range(len(iface_guids)):
+            try:
+                reg_subkey = wr.OpenKey(reg_key, iface_guids[i] + r'\Connection')
+                iface_names[i] = wr.QueryValueEx(reg_subkey, 'Name')[0]
+            except Exception, e:
+                pass
+        return iface_names
+
+    def _find_index(self, iface_name):
+        ifaces_guid = netifaces.interfaces()
+        absolute_iface_name = self._get_connection_name_from_guid(ifaces_guid)
+        try:
+            _required_iface_index = absolute_iface_name.index(iface_name)
+            _required_guid = ifaces_guid[_required_iface_index]
+            ip = netifaces.ifaddresses(_required_guid)[netifaces.AF_INET6][-1]["addr"]
+            print ip
+            return int(ip.split('%')[1])
+        except Exception,e:
+            print e
+            print "Interface %s not found" % iface_name
+            return None
+    
+    def _encode_address_port(self, addr, port):
+        port = str(port)
+        if isinstance(ipaddress.ip_address(addr), ipaddress.IPv6Address):
+            return '[' + addr + ']:' + port
+        return addr + ':' + port
+
+    @watched
+    def discoverSniffer(self):
+        sock = socket.socket(socket.AF_INET6, socket.SOCK_DGRAM)
+
+        # Define the output interface of the socket
+        ifn = self._find_index(IFNAME)
+        if ifn is None:
+            print('%s interface has not enabled IPv6.' % IFNAME)
+            return []
+        ifn = struct.pack('I', ifn)
+        sock.setsockopt(IPPROTO_IPV6, socket.IPV6_MULTICAST_IF, ifn)
+
+        # Send the request
+        sock.sendto(('sniffer').encode(), DISCOVERY_ADDR)
+
+        # Scan for responses
+        devs = set()
+        start = time.time()
+        while time.time() - start < SCAN_TIME:
+            if select.select([sock], [], [], 1)[0]:
+                addr, _ = sock.recvfrom(1024)
+                devs.add(addr)
+            else:
+                break
+
+        devs = [SimSniffer(addressofDevice=addr, channel=11) for addr in devs]
+        print(devs)
+
+        return devs
+
+    @watched
+    def startSniffer(self, channelToCapture, captureFileLocation, includeEthernet=False):
+        self.channel = channelToCapture
+        self._local_pcapng_location = captureFileLocation
+        self._remote_pcap_location = os.path.join(
+            REMOTE_SNIFFER_OUTPUT_PREFIX,
+            ''.join(random.choice(string.ascii_letters + string.digits) for _ in range(16)) + '.pcap'
+        )
+
+        self._ssh = paramiko.SSHClient()
+        self._ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        remote_ip = self.ipaddr.split('@')[1]
+        self._ssh.connect(remote_ip, port=REMOTE_PORT, username=REMOTE_USERNAME, password=REMOTE_PASSWORD)
+        
+        _, stdout, _ = self._ssh.exec_command(
+            'echo $$ && exec python3 %s -o %s -c %d' % (
```

Comment:
`echo $$` prints the PID of the current shell and the next executed command, because `exec` will replace the current shell with the arguments executed, without changing the PID.
I got the PID in this way. Perhaps there is a better way to get the PID instead.